src/status_im/ui/screens/wallet/events.cljs

```diff
@@ -18,107 +19,114 @@
                                    account-id
-                                   #(dispatch [:update-transactions-succes %])
-                                   #(dispatch [:update-transactions-fail %]))))
+                                   #(dispatch [success-event %])
+                                   #(dispatch [error-event %]))))
 
 (reg-fx
   :get-prices
   (fn [{:keys [from to success-event error-event]}]
-    (prices/get-prices
-     from
-     to
-     #(dispatch [success-event %])
-     #(dispatch [error-event %]))))
+    (prices/get-prices from
+                       to
+                       #(dispatch [success-event %])
+                       #(dispatch [error-event %]))))
 
 ;; Handlers
 
-;; TODO(oskarth): At some point we want to get list of relevant assets to get prices for
 (handlers/register-handler-fx
-  :load-prices
-  (fn [{{:keys [wallet] :as db} :db} [_ a]]
-    {:get-prices  {:from          "ETH"
-                   :to            "USD"
-                   :success-event :update-prices
-                   :error-event   :update-prices-fail}
-     :db (assoc db :prices-loading? true)}))
-
-(handlers/register-handler-fx
-  :refresh-wallet
+  :update-wallet
   (fn [{{:keys [web3 accounts/current-account-id network] :as db} :db} [_ a]]
     {:get-balance {:web3          web3
                    :account-id    current-account-id
-                   :success-event :update-balance
+                   :success-event :update-balance-success
                    :error-event   :update-balance-fail}
-     :dispatch    [:load-prices]
+     :dispatch    [:update-prices]
      :db          (assoc-in db [:wallet :balance-loading?] true)}))
 
-(defn assoc-error-message [db err]
-  (assoc-in db [:wallet :wallet/error] err))
-
-(handlers/register-handler-db
-  :wallet/clear-error-message
-  (fn [db [_]]
-    (update db :wallet dissoc :wallet/error)))
-
 (handlers/register-handler-fx
-  :refresh-transactions
+  :update-transactions
   (fn [{{:keys [accounts/current-account-id network] :as db} :db} _]
-    {:get-transactions {:account-id current-account-id
-                        :network network}
+    {:get-transactions {:account-id    current-account-id
+                        :network       network
+                        :success-event :update-transactions-success
+                        :error-event   :update-transactions-fail}
      :db               (assoc-in db [:wallet :transactions-loading?] true)}))
 
 (handlers/register-handler-db
-  :update-transactions-succes
+  :update-transactions-success
   (fn [db [_ transactions]]
     (-> db
+        (clear-error-message :transaction-update)
         (assoc-in [:wallet :transactions] transactions)
         (assoc-in [:wallet :transactions-loading?] false))))
 
 (handlers/register-handler-db
   :update-transactions-fail
   (fn [db [_ err]]
     (log/debug "Unable to get transactions: " err)
-    (-> (assoc-error-message db :error)
+    (-> db
```

Comment:
Any specific reason for this?

Reply:
for using the threading macro (readability) or for the separate error message ?

Reply:
About not using `(assoc-error-message db :error)` as initial value.

Reply:
I suppose it is a matter of style, I prefer to use the simple variable as first argument of the macro. It's macro compile time anyway.

Reply:
I generally agree with @yenda, it's clear that "I'm doing this set of transformations/operations to this one thing". Increases code clarity imo.